jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -655,3 +655,30 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
             if self.subset_definitions[index][i]['name'] == name:
                 self.subset_definitions[index][i]['value'] = new_value
                 break
+
+    def _import_spectral_regions(self, spec_region, mode=NewMode, viewer=None):
+        """
+        Method for importing a SpectralRegion object or list of SpectralRegion objects.
+
+        Parameters
+        ----------
+        spec_region : ~`specutils.SpectralRegion` object or list
```

Comment:
Your `for` loop seems to always assume `spec_region` is a list but docstring say it does not have to be. Which one is correct? If you do indeed accept `spec_region` not as a list, I think a new test case has to be added for that too.

Reply:
```
from specutils import SpectralRegion
import astropy.units as u
s1 = SpectralRegion(1*u.um, 2*u.um)
s2 = SpectralRegion(1*u.um, 2*u.um) + SpectralRegion(3*u.um, 4*u.um)
s3 = [SpectralRegion(1*u.um, 2*u.um), SpectralRegion(3*u.um, 4*u.um)]

for s in s1:
    print(s)
Spectral Region, 1 sub-regions:
  (1.0 um, 2.0 um) 

for s in s2:
    print(s)
Spectral Region, 1 sub-regions:
  (1.0 um, 2.0 um) 
Spectral Region, 1 sub-regions:
  (3.0 um, 4.0 um) 

for s in s3:
    print(s)
Spectral Region, 1 sub-regions:
  (1.0 um, 2.0 um) 
Spectral Region, 1 sub-regions:
  (3.0 um, 4.0 um) 
```
Can you think of a case where this does not work?

Reply:
What if someone gives scalar spectral region but a list of modes. How do we check?

Reply:
Can you give an example of that?

Reply:
Like this?

```python
obj._import_spectral_regions(SpectralRegion(1*u.um, 2*u.um), mode=[NewMode, NewMode])
```

Reply:
Since it is a private method, we can assume that combinations like this will not be used.

Reply:
Can you at least add a test case where `spec_region` is not a list? Thanks!

Reply:
Done